influxql/iterator.go

```diff
@@ -618,6 +618,8 @@ func (a IteratorCreators) CreateIterator(opt IteratorOptions) (Iterator, error)
 			itr, err := ic.CreateIterator(opt)
 			if err != nil {
 				return err
+			} else if itr == nil {
```

Comment:
Should we check if there were no iterators created and return nil if there are none at the bottom of this function? We still call `Merge`. I'm having a difficult time remembering the logic of the query engine at the moment, but I think it takes a `nil` return from `CreateIterator` to be "I can safely replace this with a `nilFloatIterator` at the top level. Otherwise, the `nil` is supposed to be bounced to the top so it can make that decision.

I need to take a hard look at the code though as the reason for that may not be needed anymore. I think it might have originally been done for casting.